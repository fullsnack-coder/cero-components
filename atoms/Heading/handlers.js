export const createHandlerBlur =
  ({ onBlur }) =>
  (event) => {
    onBlur(event)
  }
