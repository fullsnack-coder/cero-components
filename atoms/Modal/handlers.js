// Sync with ./Modal.module.css#L13-L17
const FADE_OUT_ANIMATION_TIME = 400

export const createHandlerClick = () => (event) => {
  event?.stopPropagation()
}

export const createHandlerClose =
  ({ onClose, setOnFadeOut }) =>
  () => {
    setOnFadeOut(true)
    setTimeout(onClose, FADE_OUT_ANIMATION_TIME)
  }

export const createHandlerSecondaryAction =
  ({ onSecondaryAction }) =>
  () => {
    onSecondaryAction()
  }
