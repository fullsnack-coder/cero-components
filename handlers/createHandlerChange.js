const createHandlerChange =
  ({ onChange }) =>
  (event) => {
    onChange(event)
  }

export default createHandlerChange
