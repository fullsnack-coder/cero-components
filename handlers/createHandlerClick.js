const createHandlerClick =
  ({ onClick }) =>
  (event) => {
    onClick(event)
  }

export default createHandlerClick
