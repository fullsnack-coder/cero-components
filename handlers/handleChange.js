const handleChange =
  ({ onChange }) =>
  (event) => {
    onChange(event)
  }

export default handleChange
