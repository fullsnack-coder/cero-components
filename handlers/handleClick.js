const handleClick =
  ({ onClick }) =>
  (event) => {
    onClick(event)
  }

export default handleClick
