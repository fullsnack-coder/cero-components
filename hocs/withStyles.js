import React from 'react'
import { getClasses } from '@glrodasz/storybook-tools-styles'

const withStyles = (styles) => (WrappedComponent) => {
  const WithStylesComponent = (props) => {
    const allProps = { ...WrappedComponent.defaultProps, ...props }
    return (
      <WrappedComponent getStyles={getClasses(styles)(allProps)} {...props} />
    )
  }

  WithStylesComponent.displayName = WrappedComponent.displayName

  return WithStylesComponent
}

export default withStyles
