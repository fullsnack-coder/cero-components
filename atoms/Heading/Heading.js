import React from 'react'
import PropTypes from 'prop-types'

import styles from './Heading.module.css'
import { options } from './constants'
import withStyles from '../../hocs/withStyles'

import { createHandlerBlur } from './handlers'

export const Heading = ({
  children,
  isCentered,
  isInline,
  isEditable,
  onBlur,
  getStyles,
}) => {
  return (
    <h1
      className={getStyles('heading', ['color', 'size', 'weight'], {
        'is-centered': isCentered,
        'is-inline': isInline,
      })}
      onBlur={createHandlerBlur({
        onBlur,
      })}
      contentEditable={isEditable}
    >
      {children}
    </h1>
  )
}

Heading.propTypes = {
  children: PropTypes.string.isRequired,
  getStyles: PropTypes.func.isRequired,
  isInline: PropTypes.bool,
  isCentered: PropTypes.bool,
  isEditable: PropTypes.bool,
  onBlur: PropTypes.func,
  color: PropTypes.oneOf(options.colors),
  size: PropTypes.oneOf(options.sizes),
  weight: PropTypes.oneOf(options.weights),
}

Heading.defaultProps = {
  color: 'base',
  size: 'md',
  weight: 'bold',
  isEditable: false,
  onBlur: () => {},
}

export default withStyles(styles)(Heading)
