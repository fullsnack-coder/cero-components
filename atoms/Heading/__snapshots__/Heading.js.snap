// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ storybook ] [ Atoms/Heading ] should render Centered 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-md weight-bold is-centered"
  >
    Lose consequence night studies Wolves should wizards destruction burglar? All right, then. Keep your secrets.
  </h1>
</DocumentFragment>
`;

exports[`[ storybook ] [ Atoms/Heading ] should render Colors 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-md weight-bold"
  />
  <h1
    class="heading color-primary size-md weight-bold"
  />
  <h1
    class="heading color-tertiary size-md weight-bold"
  />
  <h1
    class="heading color-inverted size-md weight-bold"
  />
</DocumentFragment>
`;

exports[`[ storybook ] [ Atoms/Heading ] should render Default 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-md weight-bold"
  />
</DocumentFragment>
`;

exports[`[ storybook ] [ Atoms/Heading ] should render Inline 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-md weight-bold is-inline"
  />
</DocumentFragment>
`;

exports[`[ storybook ] [ Atoms/Heading ] should render Sizes 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-xs weight-bold"
  />
  <h1
    class="heading color-base size-sm weight-bold"
  />
  <h1
    class="heading color-base size-md weight-bold"
  />
  <h1
    class="heading color-base size-lg weight-bold"
  />
  <h1
    class="heading color-base size-xl weight-bold"
  />
  <h1
    class="heading color-base size-2xl weight-bold"
  />
</DocumentFragment>
`;

exports[`[ storybook ] [ Atoms/Heading ] should render Weights 1`] = `
<DocumentFragment>
  <h1
    class="heading color-base size-md weight-normal"
  />
  <h1
    class="heading color-base size-md weight-bold"
  />
</DocumentFragment>
`;
